src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -0,0 +1,1242 @@
+            {
+                return readFunc();
+            }
+            else if (mapping.TypeDesc.FormatterName == "String")
+            {
+                if (mapping.TypeDesc.CollapseWhitespace)
+                {
+                    return CollapseWhitespace(readFunc());
+                }
+                else
+                {
+                    return readFunc();
+                }
+            }
+            else
+            {
+                string methodName = $"To{mapping.TypeDesc.FormatterName}";
+                if (!mapping.TypeDesc.HasCustomFormatter)
+                {
+                    MethodInfo method = typeof(XmlConvert).GetMethod(methodName, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
+                    return method.Invoke(null, new object[] { readFunc() });
+                }
+                else
+                {
+                        MethodInfo method = typeof(XmlSerializationReader).GetMethod(methodName, BindingFlags.Static | BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
+                        return method.Invoke(this, new object[] { readFunc() });
+                }
+            }
+        }
+
+        object WriteStructMethod(StructMapping mapping, bool isNullable, bool checkType, string defaultNamespace)
+        {
+            if (mapping.IsSoap)
+                throw new PlatformNotSupportedException();
+            else
+                return WriteLiteralStructMethod(mapping, isNullable, checkType, defaultNamespace);
+        }
+
+        object WriteNullableMethod(NullableMapping nullableMapping, bool checkType, string defaultNamespace)
+        {
+            object o = Activator.CreateInstance(nullableMapping.TypeDesc.Type);
+            if (!ReadNull())
+            { 
+                ElementAccessor element = new ElementAccessor();
+                element.Mapping = nullableMapping.BaseMapping;
+                element.Any = false;
+                element.IsNullable = nullableMapping.BaseMapping.TypeDesc.IsNullable;
+
+                WriteElement(ref o, null, element, null, false, false, false, defaultNamespace);
+            }
+            else
+            {
+                
+            }
+
+            return o;
+        }
+
+        private object WriteEnumMethod(EnumMapping mapping, Func<string> readFunc)
+        {
+            if (mapping.IsSoap)
+            {
+                throw new NotSupportedException("mapping.IsSoap");
+            }
+
+            string source = readFunc();
+
+            if (mapping.IsFlags)
+            {
+                Dictionary<string, long> table = WriteHashtable(mapping, mapping.TypeDesc.Name);
+                return Enum.ToObject(mapping.TypeDesc.Type, ToEnum(source, table, mapping.TypeDesc.Name));
+            }
+            else
+            {
+                return Enum.Parse(mapping.TypeDesc.Type, source);
```

Comment:
>Enum.Parse [](start = 23, length = 10)

Enum.Parse is VERY VERY slow. We will need to replace this with something better in the long run.